examples/jacobi.cpp

```diff
@@ -0,0 +1,274 @@
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+// IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include <cstdlib>
+#include <cstdio> 
+#include <iostream>
+
+#include "RAJA/RAJA.hpp"
+#include "RAJA/util/defines.hpp"
+
+#include "memoryManager.hpp"
+
+#define CUDA_BLOCK_SIZE 16
+
+
+/*
+  Example 3: Jacobi Method For Solving a Linear System
+
+  ----[Details]--------------------
+  This code assumes a second order finite difference 
+  spatial discretization on a lattice with unit grid spacing
+  
+  More specifically it solves:
+  -U_xx - U_yy = 1.0 inside the domain 
+  and U = 0 on the boundary of the domain
+
+  ----[RAJA Concepts]---------------
+  1. RAJA ForallN
+  2. RAJA Reduction
+  3. RAJA::omp_collapse_nowait_exec
+  3. Reducing length of RAJA statements
+*/
+
+int main(int RAJA_UNUSED_ARG(argc), char** RAJA_UNUSED_ARG(argv[]))
+{
+
+  std::cout<<"Example 3: Jacobi Method for Linear System"<<std::endl;
+
+  //----[Parameters for Solver]-------
+  double tol = 1e-5;
+  int maxIter = 100000;
+  int N = 100;
+  int NN=(N+2)*(N+2);
+  int iteration;
+  double resI2;
+
+  /*
+    invD - accumulation of finite difference coefficients 
+    f - right hand side term          
+  */
+  double invD = 1./4.0;  double f = 1.0;    
+
+  /*
+    Variables to hold approximation
+   */
+  double *I    = memoryManager::allocate<double>(NN);
+  double *Iold = memoryManager::allocate<double>(NN);
+
+  /*
+    Intialize data to zero
+  */
+  memset(I,0,NN*sizeof(double));
+  memset(Iold,0,NN*sizeof(double));
+
+  printf("Standard  C++ Loop \n");
+  resI2 = 1;  iteration = 0; 
+
+  /*
+    Carry out iterations of the Jacobi scheme until a tolerance is met
+  */
+  while(resI2>tol*tol) {
+    
+    /*
+    Iteration of the Jacobi Scheme
+    */
+    for(int n=1;n<=N;++n) {
+      for(int m=1;m<=N;++m) {
+
+        int id = n*(N+2) + m;
+                
+        I[id] = invD*(f-Iold[id-N-2]-Iold[id+N+2]-Iold[id-1]-Iold[id+1]);                
+      }
+    }
+
+    /*
+      Residual is computed via ||I - I_{old}||_{l2} + furthermore I_{old} is updated
+    */
+    resI2 = 0.0; 
+    for(int k=0; k<NN; k++) {
+      resI2 += (I[k]-Iold[k])*(I[k]-Iold[k]);
+      Iold[k]=I[k];
+    }
+    
+    if(iteration > maxIter) {        
+      printf("Standard C++ Loop - Maxed out on iterations \n");
+      exit(-1);
+    }
+
+    iteration++;
+  }
+  printf("Value at grid point ((N-1), (N-1)): %lg \n", I[N+N*(N+2)]);
+  printf("No of iterations: %d \n \n",iteration);
+    
+  /*
+    RAJA loop calls may be shortened by defining policies before hand
+  */
+  RAJA::RangeSegment jacobiRange = RAJA::RangeSegment(1,(N+1)); 
+  RAJA::RangeSegment gridRange = RAJA::RangeSegment(0,NN); 
+  using jacobiSeqNestedPolicy = RAJA::NestedPolicy< RAJA::ExecList< RAJA::seq_exec, RAJA::seq_exec > >;
```

Comment:
`RAJA::RangeSegment <name> = RAJA::RangeSegment(<args...>)` can be simplified to `RAJA::RangeSegment <name> (<args...>)`

maybe add explanation for what's required with NestedPolicy here.